frontend/docsettings.lua

```diff
@@ -60,46 +61,54 @@ local function buildCandidates(list)
                                end
                            end)
 
-    return candidates
+    -- Paths only are used
+    local candidates_path = {}
+    for _, candidate in ipairs(candidates) do
+        table.insert(candidates_path, candidate.path)
+    end
+    return candidates_path
 end
 
 --- Returns path to sidecar directory (`filename.sdr`).
 --
 -- Sidecar directory is the file without _last_ suffix.
 -- @string doc_path path to the document (e.g., `/foo/bar.pdf`)
 -- @treturn string path to the sidecar directory (e.g., `/foo/bar.sdr`)
-function DocSettings:getSidecarDir(doc_path)
+function DocSettings:getSidecarDir(doc_path, force_location)
     if doc_path == nil or doc_path == '' then return '' end
-    local file_without_suffix = doc_path:match("(.*)%.")
-    if file_without_suffix then
-        return file_without_suffix..".sdr"
+    local path = doc_path:match("(.*)%.") or doc_path -- file path without the last suffix
+    local location = force_location or G_reader_settings:readSetting("document_metadata_folder", ":doc")
+    if location == ":sidecars" then
```

Comment:
Further development might be to let a user to choose the location of sdr folders. In that case the setting would contain path to the "sidecar" storage. The leading `:` would help to distinguish standard and custom locations.
If such a development has no sense, no need to have a string for the setting, it would be boolkean and the menu would have a simple checkbox.

Reply:
A string is fine and allows for more future than a boolean.
But ordinary strings (without ":") are fine. If we ever want to include a path in there, we can require/ensure it starts or ends with a `/` so we can easily check for that.